subscription-service/src/main/kotlin/com/egm/stellio/subscription/model/Subscription.kt

```diff
@@ -51,7 +51,8 @@ data class Subscription(
     @JsonProperty(value = JSONLD_CONTEXT)
     val contexts: List<ExpandedTerm>,
     val throttling: Int? = null,
-    val lang: String? = null
+    val lang: String? = null,
+    val datasetId: List<String>? = emptyList()
```

Comment:
- nullable with a default value does not make sense
- initializing with an empty list will probably have side effect when serializing a subscription who has no datasetId (the field should not be present but it will be an empty array)

Reply:
when trying to create a subscription with no datasetId the null value causes an error before inserting in the database, that's why I changed it to an empty list

Reply:
as expected, it created side effects (the new 046_19 was failing but other subscriptions tests were failing). I fixed it.